### PR TITLE
Add Python Lambda

### LINKED ISSUE #1583: Create placeholder Python Lambda

### PR DESCRIPTION
**Pull request by lpsinger**
Fixes #1583.

Depends on architect/hydrate#181 and architect/sandbox#727.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+astropy
```
Comment (Courey) [inline requirements.txt:1]: where is this being used?

Comment (lpsinger) [reply requirements.txt:1]: It isn't yet, but it will be.

Comment (lpsinger) [reply requirements.txt:1]: Would you prefer that I remove this?

**Review by dakota002 — APPROVED** [review]
Seems straight-forward enough, though I am not familiar with FastAPI and Magnum. I did some reading on the former and I get that part now. Is [this](https://pypi.org/project/magnum/) the latter?

**Comment by lpsinger** [comment]
> Seems straight-forward enough, though I am not familiar with FastAPI and Magnum. I did some reading on the former and I get that part now.

They were new to me too, but @jak574 has used them (see #1584). The advantage of FastAPI over more common Python web frameworks like Flask or Django is that FastAPI provides an auto-generated OpenAPI spec and API browser (derived from Python type annotations) whereas Flask or Django require third-party extensions and a bunch of boilerplate to do that.

> Is [this](https://pypi.org/project/magnum/) the latter?

Yes. It is an adapter that allows you to run any Python ASGI app in a serverless context (e.g. AWS Lambda).

**Comment by lpsinger** [comment]
> Is [this](https://pypi.org/project/magnum/) the latter?

Ah, no, actually, it's [mangum](https://pypi.org/project/mangum/) (note the deliberately odd spelling).

**Comment by dakota002** [comment]
Ah now that makes more sense. I was reading it as magnum this whole time

**Comment by jak574** [comment]
> Seems straight-forward enough, though I am not familiar with FastAPI and Magnum. I did some reading on the former and I get that part now. Is [this](https://pypi.org/project/magnum/) the latter?

It's "Mangum" not Magnum. https://mangum.io

Basically a ASGI layer to allow FastAPI to work with AWS Lambda.

**Pull request merged**